clis/teliod/src/core_api.rs

```diff
@@ -0,0 +1,173 @@
+                if BASE64_STANDARD
+                    .decode(recvd_pk)?
+                    .as_slice()
+                    .cmp(&public_key.0)
+                    .is_eq()
+                {
+                    if let Some(identifier) = node.get("identifier").and_then(|i| i.as_str()) {
+                        info!("Match found! Identifier: {}", identifier);
+                        return Ok(identifier.to_owned());
+                    }
+                }
+            }
+        }
+    }
+    Err(Error::FetchingIdentifierError(status))
+}
+
+pub async fn update_machine(client_config: &ClientConfig) -> Result<StatusCode, Error> {
+    debug!("Updating machine");
+    let client = Client::new();
+    Ok(client
+        .patch(&format!(
+            "{}/meshnet/machines/{}",
+            API_BASE,
+            client_config.machine_identifier.clone()
+        ))
+        .header(
+            header::AUTHORIZATION,
+            format!("Bearer token:{}", &client_config.auth_token),
+        )
+        .header(header::CONTENT_TYPE, "application/json")
+        .header(header::ACCEPT, "application/json")
+        .json(&MeshConfig {
+            public_key: client_config.public_key,
+            hardware_identifier: client_config.hw_identifier.clone(),
+            os: OS_NAME.to_owned(),
+            os_version: "unknown".to_owned(),
+            device_type: "other".to_owned(),
+            traffic_routing_supported: true,
+        })
+        .send()
+        .await?
+        .status())
+}
+
+pub async fn get_meshmap(client_config: Arc<ClientConfig>) -> Result<MeshMap, Error> {
+    debug!("Getting meshmap");
+    let client = Client::new();
+    Ok(serde_json::from_str(
+        &client
+            .get(&format!(
+                "{}/meshnet/machines/{}/map",
+                API_BASE, client_config.machine_identifier
+            ))
+            .header(
+                header::AUTHORIZATION,
+                format!("Bearer token:{}", client_config.auth_token),
+            )
+            .header(header::ACCEPT, "application/json")
+            .send()
+            .await?
+            .text()
+            .await?,
+    )?)
+}
+
+pub async fn register_machine(
+    hw_identifier: &String,
+    public_key: PublicKey,
+    auth_token: &String,
+) -> Result<MachineIdentifier, Error> {
+    info!("Registering machine");
+    let client = Client::new();
+    let response = client
+        .post(&format!("{}/meshnet/machines", API_BASE))
+        .header(
+            header::AUTHORIZATION,
+            format!("Bearer token:{}", auth_token),
+        )
+        .header(header::CONTENT_TYPE, "application/json")
+        .header(header::ACCEPT, "application/json")
+        .json(&MeshConfig {
+            public_key: public_key,
+            hardware_identifier: hw_identifier.clone(),
+            os: OS_NAME.to_owned(),
+            os_version: "unknown".to_owned(),
```

Comment:
We could set this or `device_type` to `teliod`, so that we could quickly distinguish it in the list, like we do in tclid: `"os_version": "macos tcli"`

Reply:
`device_type` cant be `teliod`. It is not a valid expected value. `os_version` could be changed yes!